includes/RottenLinksLuaLibrary.php

```diff
@@ -0,0 +1,58 @@
+<?php
+namespace Miraheze\RottenLinks;
+
+use MediaWiki\Extension\Scribunto\Engines\LuaCommon\LibraryBase;
+use MediaWiki\Extension\Scribunto\Engines\LuaCommon\LuaEngine;
+use MediaWiki\Extension\Scribunto\Engines\LuaCommon\LuaError;
+use MediaWiki\MediaWikiServices;
+use Wikimedia\Rdbms\ILoadBalancer;
+
+class RottenLinksLuaLibrary extends LibraryBase {
+
+	private ILoadBalancer $loadBalancer;
+
+	/**
+	 * @param LuaEngine $engine
+	 */
+	public function __construct( $engine ) {
+		parent::__construct( $engine );
+		// Unfortunately, Scribunto currently does not offer us any options to do
+		// dependency injection, so we have to pretend that we do. Luckily, there
+		// is already an upstream task: https://phabricator.wikimedia.org/T375835
+		$this->loadBalancer = MediaWikiServices::getInstance()->getDBLoadBalancer();
+	}
+
+	/**
+	 * @param mixed $url
+	 * @return array
+	 * @internal
+	 */
+	public function onGetStatus( $url = null ) {
```

Comment:
Param type: No: https://github.com/miraheze/RottenLinks/pull/81/commits/056c1417b7a46b11d6c08d29fc7366ac0bb3a4b9
Default: Required in case the user types in `mw.ext.rottenLinks.getStatus()` (Scribunto won't give us an argument, so we pretend it's null)
Return type: I'll see

Reply:
Can't you type hint mixed?